access-token-management/src/AccessTokenManagement.OpenIdConnect/Implementation/OpenIdConnectClientAccessTokenRetriever.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Duende Software. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using Duende.AccessTokenManagement.OTel;
+using Microsoft.AspNetCore.Http;
+
+namespace Duende.AccessTokenManagement.OpenIdConnect.Implementation;
+
+/// <summary>
+/// TokenRetriever that implements the logic on behalf of <see cref="AccessTokenHandler{TTokenRetriever,TToken}"/> to
+/// retrieve a ClientCredentials Token for an OpenID Connect flow.
+/// </summary>
+/// <param name="httpContextAccessor"></param>
+/// <param name="parameters"></param>
+internal class OpenIdConnectClientAccessTokenRetriever(
```

Comment:
Isn't this obtaining a user token?

```suggestion
/// <summary>
/// An <see cref="ITokenRetriever" /> implementation that retrieves a user access token.
/// </summary>
/// <param name="httpContextAccessor"></param>
/// <param name="parameters"></param>
internal class OpenIdConnectClientAccessTokenRetriever(
```

Reply:
No, it's a client credential token. I find this handler a bit dubious, as it's 95 the same as the plain client credentials handler, but I wanted to keep 100% feature parity.